cpp/src/parquet/column_reader.cc

```diff
@@ -1117,11 +1124,10 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatch(int64_t batch_size, int16_t* def
   int64_t num_def_levels = 0;
   int64_t values_to_read = 0;
   ReadLevels(batch_size, def_levels, rep_levels, &num_def_levels, &values_to_read);
-
   *values_read = this->ReadValues(values_to_read, values);
+  ARROW_DCHECK_GE(values_to_read, *values_read);
```

Comment:
Can this occur on a corrupted file? If so, it should be an exception.

Reply:
Yeah, you're right, this can occur on a corrupt file!

Reply:
Oh-no, we cannot...If `values_read` is too less, it might be a corrupt file, but when `values_to_read < *values_read`, `ReadValues` returns too many...